src/components/signupform/signupform.js

```diff
@@ -1,147 +1,319 @@
 import React, { Component } from "react";
 import { Form, Button } from "react-bootstrap";
-import "./signupform.scss";
-import { MdBorderAll } from "react-icons/md";
+import { connect } from "react-redux";
+import { signup } from "../../reducers/userSlice";
 
 class SignUpForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      username: "",
-      isLogin: true,
+      firstName: "",
+      firstNameError: "",
+      lastName: "",
+      lastNameError: "",
       email: "",
+      emailError: "",
       password: "",
-      passwordVerify: "",
+      passwordError: "",
+      verifyPassword: "",
+      verifyPasswordError: "",
+      phone: "",
+      phoneError: "",
+      shortDescription: "",
+      shortDescriptionError: "",
+      isFormInvalid: true,
+      formSubmissionError: "",
     };
   }
 
-  handleChange = (params, event) => {
-    event.preventDefault();
-    switch (params) {
-      case "email":
-        this.setState({ email: event.target.value });
+  onFieldChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({ [name]: value }, () => {
+      this.checkFieldValidation(name);
+    });
+  };
+
+  checkFieldValidation = (field) => {
```

Comment:
How about making `checkFieldValidation(field)` a module and use wherever we need?

Reply:
Good idea; but same fields are having different validations for different models. For example, somewhere shortDescription is required to be only 3 characters long, but somewhere else it has to be atleast 5 characters

Reply:
I would say standardize as much as you can if possible!
@vaibhavdaren, what do you think?

Reply:
Okay, I can think of prefixed names to differentiate it. But one query, when calling the moduralised function we will have to bind "this" reference to it first for every component which will eventually return a new function and we will calling that new function for that component. So, if the bind() method will end up creating a new function anyway; are we gaining something by moduralising this function?

Reply:
Resolved in #26